go2eiko.py

```diff
@@ -0,0 +1,336 @@
+        excepted_files=5,
+    )
+
+
+def download_characters():
+    
+    # Character's base assets
+    download_recursively(
+        path = "libs/gi/assets/src/gen/chars",
+        output_path="gamedata/assets/characters/",
+        name_converter="index.ts",
+        force_download=FORCE,
+        expected_files=15,
+    )
+    
+    # Character namecards
+    def name_converter(old_filename: str) -> str:
+        ext = old_filename.split(".")[-1]
+        return f"card.{ext}"
+    
+    def output_path(old_filename: str) -> str:
+        pattern = "Character_([A-Za-z_]+)_Card\.[a-z]+"
+        character_name = re.search(pattern, old_filename).group(1).lower()
+        return os.path.join("gamedata/assets/characters", character_name)
+    
+    download_folder(
+        path="libs/gi/char-cards/src",
+        output_path=output_path,
+        name_converter=name_converter,
+    )
+    
+    
+
+
+def download_weapons():
+
+    def name_converter(old_filename: str) -> str:
+        
+        # Get the extension
+        ext = old_filename.split(".")[-1]
+
+        # Check in which case we are
+        if "Awaken" in old_filename:
+            return f"weapon-awaken.{ext}"
+        else:
+            return f"weapon.{ext}"
+    
+    download_recursively(
+        path = "libs/gi/assets/src/gen/weapons",
+        output_path="gamedata/assets/weapons/",
+        name_converter=name_converter,
+        force_download=FORCE,
+        expected_files=15,
+    )
+
+
+def main(**kwargs):
+
+    os.makedirs(MASTER_OUTPUT_PATH, exist_ok=True)
+
+    if kwargs["characters"]:
+        download_characters()
+
+    if kwargs["artifacts"]:
+        download_artifacts()
+    
+    if kwargs["weapons"]:
+        download_weapons()
+
+
+if __name__ == "__main__":
+
+    # Configure logging
+    logging.basicConfig(
+        level=logging.INFO,
+        format="[%(asctime)s] [%(levelname)-8s] --- %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+    # Setting up the parser
+    parser = argparse.ArgumentParser(
+        description="Download the assets from Genshin Optimizer"
+    )
+    parser.add_argument("--characters", "-c", action="store_true", help="Download the characters' assets")
+    parser.add_argument("--weapons", "-w", action="store_true", help="Download the weapons' assets")
+    parser.add_argument("--artifacts", "-a", action="store_true", help="Download the artifacts' assets")
+    parser.add_argument("--force", "-f", action="store_true", help="Force the redownload of all the assets")
+    parser.add_argument("--output", "-o", type=str, help="Path of the 'data' folder", default="./data/")
+    kwargs = vars(parser.parse_args())
+    MASTER_OUTPUT_PATH = kwargs["output"]
+    FORCE = kwargs["force"]
+
+    # Logging to GitHub's API
+    dotenv_path = os.path.join(os.path.dirname(__file__), '.env')
+    load_dotenv(dotenv_path)
+    username = os.environ.get('GITHUB_USERNAME')
+    token = os.environ.get('GITHUB_TOKEN')
```

Comment:
if we're not using dotenv later on, these variables should be named something like "EIKONOMIYA_GITHUB_STUFF"

Reply:
Changed in ed9428c16dbeea8794ec887d62dab078968c52fe (see also https://github.com/eikofee/eikonomiya/pull/28#discussion_r1554842576)